### PR TITLE
Development

### PR DESCRIPTION
**Pull request by HiteshChoudhary0098**
Created action.js file to structure the actions
Created test cases for testing basic actions

### CODE REVIEW

File: action.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+function add() {
+    return {
+        type: "OPERATOR",
+        // define a function accordingly
+        payload: (x) => x
```
Comment (vanshuhassija) [inline action.js:5]: Why is type operator? type should be something like ADD

Comment (vanshuhassija) [inline action.js:5]: Why is payload a function. Payload usually is a simple object with data.

File: action.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    }
+}
+function equal() {
+    return {
+        type: "EQUAL"
+    }
+}
+function type() {
+    return {
+        type: "TYPE"
```
Comment (vanshuhassija) [inline action.js]: What is this type for?

File: actions.test.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    expect(action.add).not.toBe(undefined)
+})
+test("add action should be function ", () => {
+    expect(typeof (action.add)).toBe("function")
+})
+test("add action should  have payload defined", () => {
+    expect(action.add().payload).not.toBe(undefined)
+})
+test("add action should  have payload defined as function", () => {
+    expect(typeof (action.add().payload)).not.toBe("function")
```
Comment (vanshuhassija) [inline actions.test.js]: Payload will be an object with data

File: actions.test.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+})
+
+// testing sub operation
+test("sub action should be defined", () => {
```
Comment (vanshuhassija) [inline actions.test.js]: sub => subtract

File: actions.test.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    expect(action.sub).not.toBe(undefined)
+})
+test("sub action should be function ", () => {
```
Comment (vanshuhassija) [inline actions.test.js]: sub=>subtract

File: actions.test.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+})
+test("sub action should  have payload defined", () => {
+    expect(action.sub().payload).not.toBe(undefined)
+})
+test("sub action should  have payload defined as function", () => {
+    expect(typeof (action.sub().payload)).not.toBe("function")
+})
+
+// testing mul operation
+test("mul action should be defined", () => {
```
Comment (vanshuhassija) [inline actions.test.js]: mul=>product everywhere

File: actions.test.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+})
+test("mul action should  have payload defined", () => {
+    expect(action.mul().payload).not.toBe(undefined)
+})
+test("mul action should  have payload defined as function", () => {
+    expect(typeof (action.mul().payload)).not.toBe("function")
+})
+
+// testing div operation
+test("div action should be defined", () => {
```
Comment (vanshuhassija) [inline actions.test.js]: What is this div action?

File: actions.test.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+
+// testing changeSign action
+test("changeSign action should be defined", () => {
+    expect(action.changeSign).not.toBe(undefined)
+})
+test("changeSign action should be function ", () => {
+    expect(typeof (action.changeSign)).toBe("function")
+})
+
```
Comment (vanshuhassija) [inline actions.test.js:256]: No test cases written for checking payload data. Test for what all properties you are expecting in payload data

**Comment by HiteshChoudhary0098** [comment]
the type function is a bit of confusing !!

**Comment by HiteshChoudhary0098** [comment]
function digitInput{
type:"DIGIT_INPUT"
}

will it be fine
@vanshuhassija

**Review by vanshuhassija — COMMENTED** [review]
please review comments

**Pull request closed without merging**